runner/runner.go

```diff
@@ -41,26 +42,34 @@ func Run(router http.Handler, conf *config.Configuration) {
 		}
 		fmt.Println("Started Listening for TLS connection on " + addr)
 		go func() {
-			listener := startListening(addr, conf.Server.KeepAlivePeriodSeconds)
+			listener := startListening(network, addr, conf.Server.KeepAlivePeriodSeconds)
 			log.Fatal(s.ServeTLS(listener, conf.Server.SSL.CertFile, conf.Server.SSL.CertKey))
 		}()
 	}
-	addr := fmt.Sprintf("%s:%d", conf.Server.ListenAddr, conf.Server.Port)
+	network, addr := listenAddrParse(conf.Server.ListenAddr, conf.Server.Port)
 	fmt.Println("Started Listening for plain HTTP connection on " + addr)
 	server := &http.Server{Addr: addr, Handler: httpHandler}
-
-	log.Fatal(server.Serve(startListening(addr, conf.Server.KeepAlivePeriodSeconds)))
+	log.Fatal(server.Serve(startListening(network, addr, conf.Server.KeepAlivePeriodSeconds)))
 }
 
-func startListening(addr string, keepAlive int) net.Listener {
+func startListening(network, addr string, keepAlive int) net.Listener {
 	lc := net.ListenConfig{KeepAlive: time.Duration(keepAlive) * time.Second}
-	conn, err := lc.Listen(context.Background(), "tcp", addr)
+	conn, err := lc.Listen(context.Background(), network, addr)
 	if err != nil {
 		log.Fatalln("Could not listen on", addr, err)
 	}
 	return conn
 }
 
+func listenAddrParse(ListenAddr string, Port int) (string, string) {
+	if strings.HasPrefix(ListenAddr, "unix:") {
+		path := strings.TrimPrefix(ListenAddr, "unix:")
+		os.Remove(path)
```

Comment:
Only removing the file seems faulty, i think when calling .Close on the net.Listener the socket file should be automatically removed. I think I'd prefer if we'd do a graceful shutdown and properly close the connection. What do you think?

Reply:
I added defer close(), however, it does not remove file in my tests. It could be im `ctrl + c` the process 🤷🏻

Reply:
Even within the Gin implementation they remove the socket on defer `os.remove`

https://github.com/gin-gonic/gin/blob/53fbf4dbfbf465b552057e6f8d199a275151b7a1/gin.go#L511-L529

We can just do the same